File: Allocation.py
b,x,y,z = map(int, input().split())
x1 =list( map(int, input().split()))
y1 =list( map(int, input().split()))
z1 =list( map(int, input().split()))
x1.sort()
y1.sort()
z1.sort()
int d=0
for i in range(0,x):
    for j in range(0,y):
        for k in range(0,z):
            sum=x1[i]+y1[j]+z1[k]
                if(sum>d & sum<b):
                     d=sum
                
   
 print(b-d)
    
}

File: Allocation.java
import java.util.*;


public class Allocation{
    public static void main(String[] args) {
        Scanner sc =new Scanner(System.in);
        int n=sc.nextInt();
        int p=1;
        while(n--!=0)
        {
            int a=sc.nextInt();
            int b=sc.nextInt();
            int a1[]=new int[a];
            for(int i=0;i<a;i++){
                a1[i] = sc.nextInt();
            }
            Arrays.sort(a1);
            int sum=a1[0],i=1,count=0;
            while(sum<=b)
            {
                sum=sum+a1[i];
                i++;
                count++;

            }
            System.out.println("Case #"+p+": "+count);
            p++;
        }
        
    }
}